src/abstractdatatable/join.jl

```diff
@@ -94,23 +108,17 @@ function update_row_maps!(left_table::AbstractDataTable,
     # helper functions
     @inline update!(ixs::Void, orig_ix::Int, join_ix::Int, count::Int = 1) = nothing
     @inline function update!(ixs::RowIndexMap, orig_ix::Int, join_ix::Int, count::Int = 1)
-        for i in 1:count
-            push!(ixs.orig, orig_ix)
-        end
-        for i in join_ix:(join_ix+count-1)
-            push!(ixs.join, i)
-        end
+        append!(ixs.orig, fill(orig_ix, count))
```

Comment:
Damn, it's time to merge this PR, I'm more and more lost. Then the `sizehint!` isn't needed at all, `resize!` is enough.